pages/index.js

```diff
@@ -1,7 +1,28 @@
+// lib imports
+import useSWR from "swr";
+import styled from "styled-components";
+// components
+import Error from "@/components/Error";
+import Loading from "@/components/Loading";
+import EventList from "@/components/EventList";
+import Header from "@/components/Header";
+
+const StyledMain = styled.main`
+  height: 100vh;
+  padding-block: 3rem;
+`;
+
 export default function HomePage() {
+  const { data: events, isLoading, error } = useSWR(`/api/events`);
+
   return (
-    <div>
-      <h1>Hello from Next.js</h1>
-    </div>
+    <>
+      <Header />
+      <StyledMain>
+        {events && <EventList events={events} />}
+        {isLoading && <Loading />}
+        {error && <Error>{"Oops something went wrong"}</Error>}
```

Comment:
```suggestion
        {error && <Error>{error.message}</Error>}
```
Would be maybe a bit more understandable for the user, what went wrong. But you can try it out!?